openmrs/omod/src/main/java/org/projectbuendia/openmrs/web/controller/ProfileManager.java

```diff
@@ -145,68 +173,79 @@ String getNextVersionedFilename(String name) {
-    // TODO: Give some error message if the profile is not uploaded correctly
-    //       (When tomcat7 didn't have permission to PROFILES_DIR it was failing
-    //        without showing any error message to the user)
-    void addProfile(MultipartHttpServletRequest request, ModelMap model) {
+    private void addProfile(MultipartHttpServletRequest request, ModelMap model) {
         List<String> lines = new ArrayList<>();
         MultipartFile mpf = request.getFile("file");
         if (mpf != null) {
+            String message = "";
+            String filename = mpf.getOriginalFilename();
             try {
                 File tempFile = File.createTempFile("profile", null);
                 mpf.transferTo(tempFile);
-                if (execute(VALIDATE_CMD, tempFile, lines)) {
-                    String filename = getNextVersionedFilename(mpf.getOriginalFilename());
-                    File newFile = new File(PROFILE_DIR, filename);
-                    model.addAttribute("filename", filename);
+                boolean exResult = execute(VALIDATE_CMD, tempFile, lines);
+                if (exResult) {
+                    filename = getNextVersionedFilename(filename);
+                    File newFile = new File(profileDir, filename);
                     FileUtils.moveFile(tempFile, newFile);
-                    model.addAttribute("success", "add");
+                    model.addAttribute("success", true);
+                    message = "Success adding profile: ";
                 } else {
-                    model.addAttribute("failure", "add");
-                    model.addAttribute("filename", mpf.getOriginalFilename());
-                    model.addAttribute("output", StringUtils.join(lines, "\n"));
+                    model.addAttribute("success", false);
+                    message = "Error adding profile: ";
                 }
             } catch (Exception e) {
+                model.addAttribute("success", false);
+                message = "Problem saving uploaded profile: ";
+                lines.add(e.getMessage());
                 log.error("Problem saving uploaded profile", e);
+            } finally {
+                model.addAttribute("operation", "add");
+                model.addAttribute("message", message + filename);
+                model.addAttribute("filename", filename);
+                model.addAttribute("output", StringUtils.join(lines, "\n"));
             }
         }
     }
 
     /** Applies a profile to the OpenMRS database. */
-    void applyProfile(File file, ModelMap model) {
+    private void applyProfile(File file, ModelMap model) {
         List<String> lines = new ArrayList<>();
         if (execute(APPLY_CMD, file, lines)) {
             setCurrentProfile(file.getName());
-            model.addAttribute("success", "apply");
+            model.addAttribute("success", true);
+            model.addAttribute("message", "Success applying profile: " + file.getName());
```

Comment:
The entire OpenMRS side of things aren't localized so I don't think it applies here.